src/main/java/org/opensearch/neuralsearch/search/MultiLeafFieldComparator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search;
+
+import java.io.IOException;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.LeafFieldComparator;
+import org.apache.lucene.search.Scorable;
+
+/*
+MultiLeafFieldComparator holds information when sort criteria is applied on more that one field.
+This class is same as of lucene. Because lucene implementation does not have public access we have to add it in neural search plugin.
+https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/MultiLeafFieldComparator.java
```

Comment:
this is exact copy of Lucene file? if yes then we might need to reconsider what we are trying to do here.

@vibrantvarun if there are classes which are exact copy from Lucene with access modifiers changed then I would probably put them in package with namespace of Lucene in it. This way atleast we know what are the classes which are from Lucene